examples/build_adsorbate_place_and_rotate.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+
+"""
+Example script showing how to add a hydrogen to a zeolite cluster,
+then add and rotate an ethanol molecule to the desired position.
+"""
+
+def test_adsorbate_placer():
+    from ase.io import read
+    from ase.build import molecule
+    import numpy as np
+    from carmm.build.adsorbate_placer import place_adsorbate, rotate_and_place_adsorbate
+    from ase import Atoms
+
+    molecule = molecule('CH3CH2OH')
+    h_atom = Atoms('H', positions=[(0, 0, 0)])
+
+    site = read("data/H-Y_cluster/H-Y_cluster.xyz")
+
+    zeolite, rotated_ads = place_adsorbate(h_atom, site, 0, 0, 1.0)
+
+    ads_and_site, rotated_ads = rotate_and_place_adsorbate(molecule, zeolite, 1.0,
+                                                           2, 0, 0,
+                                                           rotation=[45, 0, -45])
+
+    comp_pos1 = np.array([2.00311656e+01, 5.31509397e+00, 1.89702619e+00])
+    comp_pos2 = np.array([2.04405930e+01, 4.53669268e+00, 2.54385921e+00])
+
+    error_pos1 = np.linalg.norm(comp_pos1 - rotated_ads.positions[0], axis=-1)
+    error_pos2 = np.linalg.norm(comp_pos2 - rotated_ads.positions[-1], axis=-1)
+
+    assert (np.isclose(error_pos1, 0, rtol=0, atol=1e-06))
+    assert (np.isclose(error_pos2, 0, rtol=0, atol=1e-06))
```

Comment:
This currently returns an assertion error

Reply:
Just to confirm, I am using ase==3.22.1 and numpy==1.22.3, what about you?

Reply:
Python==3.7.12
ASE==3.22.1
numpy==1.21.5 

No massive version differences. Pushed some basic printing commands into the assertion, so it should display just how badly we're off by. If it's say, 1e-05, the tolerance might be too tight. But anything larger than 1e-04 is cause for concern.

Reply:
The problem persists on my end, also in `view()` hydrogen index=3 and hydrogen index=9 are less than 0.5A apart, which would result in a failure in initialisation of FHI-aims calculation

Reply:
> Python==3.7.12 ASE==3.22.1 numpy==1.21.5
> 
> No massive version differences. Pushed some basic printing commands into the assertion, so it should display just how badly we're off by. If it's say, 1e-05, the tolerance might be too tight. But anything larger than 1e-04 is cause for concern.

I am on Python 3.10 on my local machine

Reply:
Hmm. Problematic. What are the error values printed out?

Reply:
AssertionError: Error = 3.29883926784293

Reply:
Aaaaah, now that is sizeable. Hmm. Let me make a new conda environment and test this out. Apologies about this.

Reply:
I see some dependency on `from carmm.analyse.neighbours import neighbours`, have you modified it at all?

Reply:
I haven't modified any existing files to my knowledge. 

However, I rolled back to Python 3.1 and am getting the same error you observed - this doesn't happen with Python 3.7. Very curious. Will get back to you with the part of the code which is misbehaving.

Reply:
For some reason beyond the ken of man - the order of the Neighbour positions lists changes between the two versions. So the code itself is working as intended, but reversing some rotations. Now put some stop gap statements to preserve order, but will tell you when everything is working.

Reply:
Yeah, without ordering the neighbour list, python3.7 by default orders the neighbours [3,2], while 3.1 orders them [2,3]. A strange one, but the work around is simple.